services_app/src/androidTest/java/org/opendatakit/webkitserver/service/OdkWebserverServiceTest.java

```diff
@@ -307,12 +302,447 @@ private void assertResponseIsNotEmpty(File fileLocation) {
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Got an Exception when trying to use the web server: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void testServingFileWithNoExtension() {
+        // Arrange
+        File directoryLocation = createTestDirectory();
+        File fileLocation = createTestFile(directoryLocation, TEST_FILE_NAME_NO_EXTENSION, "Content with no extension");
+
+        // Act
+        IWebkitServerInterface serviceInterface = getIWebkitServerInterface();
+        restartService(serviceInterface);
+
+        // Assert
+        assertResponseForNoExtension(fileLocation);
+    }
+
+    private void assertResponseForNoExtension(File fileLocation) {
+        try {
+            Uri uri = Uri.parse(buildTestUrl(fileLocation));
+
+            HttpURLConnection connection = (HttpURLConnection) new URL(uri.toString()).openConnection();
+            int responseCode = connection.getResponseCode();
+
+            assertEquals("Response code should be HTTP_OK", HttpStatus.SC_OK, responseCode);
+
+            // Additional assertions for handling files with no extension, if needed
+        } catch (IOException e) {
+            e.printStackTrace();
+            fail("Got an IOException when trying to use the web server: " + e.getMessage());
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Got an Exception when trying to use the web server: " + e.getMessage());
+        }
+    }
+
+    @Test
+    public void testServingFileWithDifferentExtension() {
+        // Arrange
+        File directoryLocation = createTestDirectory();
+        File fileLocation = createTestFile(directoryLocation, TEST_FILE_NAME_DIFFERENT_EXTENSION, "Content with a different extension");
+
+        // Act
+        IWebkitServerInterface serviceInterface = getIWebkitServerInterface();
+        restartService(serviceInterface);
+
+        // Assert
+        assertResponseForDifferentExtension(fileLocation);
+    }
+
+    private void assertResponseForDifferentExtension(File fileLocation) {
+        try {
+            Uri uri = Uri.parse(buildTestUrl(fileLocation));
+
+            HttpURLConnection connection = (HttpURLConnection) new URL(uri.toString()).openConnection();
+            int responseCode = connection.getResponseCode();
+
+            assertEquals("Response code should be HTTP_OK", HttpStatus.SC_OK, responseCode);
+
+            // Additional assertions for handling files with different extensions, if needed
+        } catch (IOException e) {
+            e.printStackTrace();
+            fail("Got an IOException when trying to use the web server: " + e.getMessage());
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Got an Exception when trying to use the web server: " + e.getMessage());
+        }
+    }
+
+    private File createTestFile(File directoryLocation, String fileName, String content) {
```

Comment:
you still need to switch to the URI format when creating the test files. It is fine to leave the interface returning a file however, as it abstracts the file class from intefacing with the operating system but still uses it at a high level so less code needs to be revised.